expression/reconstructor/reconstructor.go

```diff
@@ -37,13 +42,16 @@ const maxBytesInterpretedAsNumber = 15
 type ExprReconstructor struct{}
 
 // Reconstruct will return the string representation of the provided value
-func (er *ExprReconstructor) Reconstruct(value []byte, hint ExprReconstructorHint) string {
+func (er *ExprReconstructor) Reconstruct(value []byte, hint ExprReconstructorHint, bech32Addr ...bool) string {
```

Comment:
Don't add more arguments to the method, add configs to the ExprReconstructor if you need to.

A simple boolean flag of whether or not to attempt bech32 de-conversion I think is enough for now.

If you want to implement the "already seen" expressions, make something more general.